tests/test_individual.py

```diff
@@ -163,13 +172,32 @@ def test_populate_ipf(self, dir_params, expected_cols):
         """
 
         # Call populate_ipf for testing
-        df = Individual.populate_ipf(1, dir_params)
 
-        assert list(df[
-                        0].get_features().index) == expected_cols, \
-            'Columns in the dataframe returned by populate_ipf() are incorrect'
+        df = Individual.populate_ipf(1, dir_params, rng=np.random.default_rng(seed))
+
+        assert (
+            list(df[0].get_features().index) == expected_cols
+        ), "Columns in the dataframe returned by populate_ipf() are incorrect"
 
         assert np.all(
             (df[0].get_features() == 0) | (df[0].get_features() == 1)
-        ).all(), 'Values in the dataframe returned ' \
-                 'by populate_ipf should be either 0 or 1'
+        ).all(), (
+            "Values in the dataframe returned "
+            "by populate_ipf should be either 0 or 1"
+        )
+
+    def test_actions_when_positive(self, dir_params, lockdown, seed):
+        """
+        Test that modify_policy_when_infected returns 'stay at home' action
+        """
+        policy = Hypothesis.read_hypotheses(dir_params, set([lockdown]), "lockdown")
+        agent = Individual.populate_ipf(1, dir_params, rng=np.random.default_rng(seed))
+        lockdown_new = agent[0].modify_policy_when_infected(policy[lockdown])
+        _, action_probs = agent[0].choose_actions_on_lockdown(lockdown_new)
+        action_out = agent[0].get_actions()
+        assert action_out[0] == "stay_at_home", (
+            f"Expected 'stay at home' got {action_out[0]}"
+        )
+        assert action_probs.round()[2] == 1, (
+            f"Expected Probability ~ 1 got {action_probs.round()[2]}"
+        )
```

Comment:
I am not sure if I understand correctly why the `round()` call is necessary here, but this might be another use case for `pytest.approx()`:

```suggestion
        assert action_probs.round()[2] == pytest.approx(1), (
            f"Expected Probability ~ 1 got {action_probs[2]}"
        )
```